src/natcap/invest/urban_nature_access.py

```diff
@@ -0,0 +1,315 @@
+            will take place in the current process.
+        args['lulc_raster_path'] (string): (required) A string path to a
+            GDAL-compatible land-use/land-cover raster containing integer
+            landcover codes.
+        args['lulc_attribute_table'] (string): (required) A string path to a
+            CSV with the following columns:
+
+            * ``lucode``: the integer landcover code represented.
+            * ``greenspace``: ``0`` or ``1`` indicating whether this landcover
+              code is (``1``) or is not (``0``) a greenspace pixel.
+
+        args['population_raster_path'] (string): (required) A string path to a
+            GDAL-compatible raster where pixels represent the population of
+            that pixel.
+        args['admin_unit_vector_path'] (string): (required) A string path to a
+            GDAL-compatible vector containing polygon administrative
+            boundaries.
+        args['greenspace_demand'] (number): (required) A positive, nonzero
+            number indicating the required greenspace, in m² per capita.
+        args['search_radius'] (number): (required) A positive, nonzero number
+            indicating the maximum distance that people travel for recreation.
+
+    Returns:
+        ``None``
+    """
+    LOGGER.info('Starting Urban Nature Access Model')
+
+    output_dir = os.path.join(args['workspace_dir'], 'output')
+    intermediate_dir = os.path.join(args['workspace_dir'], 'intermediate')
+    utils.make_directories([output_dir, intermediate_dir])
+
+    file_suffix = utils.make_suffix_string(args, 'results_suffix')
+    file_registry = utils.build_file_registry(
+        [(_OUTPUT_BASE_FILES, output_dir),
+         (_INTERMEDIATE_BASE_FILES, intermediate_dir)],
+        file_suffix)
+
+    work_token_dir = os.path.join(intermediate_dir, '_taskgraph_working_dir')
+    try:
+        n_workers = int(args['n_workers'])
+    except (KeyError, ValueError, TypeError):
+        # KeyError when n_workers is not present in args
+        # ValueError when n_workers is an empty string.
+        # TypeError when n_workers is None.
+        n_workers = -1  # Synchronous execution
+    graph = taskgraph.TaskGraph(work_token_dir, n_workers)
+
+    # Align the population raster to the LULC.
+    lulc_raster_info = pygeoprocessing.get_raster_info(
+        args['lulc_raster_path'])
+
+    population_alignment_task = graph.add_task(
+        _resample_population_raster,
+        kwargs={
+            'source_population_raster_path': args['population_raster_path'],
+            'target_population_raster_path': file_registry[
+                'resampled_population'],
+            'target_pixel_size': lulc_raster_info['pixel_size'],
+            'target_bb': lulc_raster_info['bounding_box'],
+            'target_projection_wkt': lulc_raster_info['projection_wkt'],
+            'working_dir': intermediate_dir,
+        },
```

Comment:
I like the named parameter approach. Did you choose this over regular arguments for any particular reason?

Reply:
The main reason I chose to do this was because I'm a bit rusty!  But in general, adding the keyword argument names here just felt more readable to me while I'm getting back into the swing of things.